src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/TransactionValidator.cs

```diff
@@ -42,6 +68,75 @@ public static void ValidateTransactionBundle(Hl7.Fhir.Model.Bundle bundle)
             }
         }
 
+        private static string BuildRequestUrlForConditionalQueries(EntryComponent entry, string conditionalCreateQuery)
+        {
+            return conditionalCreateQuery == null ? entry.Request.Url : entry.Request.Url + "?" + conditionalCreateQuery;
+        }
+
+        private async Task<string> GetResourceId(EntryComponent entry)
+        {
+            if (entry.Request.IfNoneExist == null && !entry.Request.Url.Contains("?", StringComparison.InvariantCulture))
+            {
+                if (entry.Request.Method == HTTPVerb.POST)
+                {
+                    return entry.FullUrl;
+                }
+
+                return entry.Request.Url;
+            }
+            else
+            {
+                string resourceType = null;
+                StringValues conditionalQueries;
+                HTTPVerb requestMethod = (HTTPVerb)entry.Request.Method;
+                bool conditionalCreate = requestMethod == HTTPVerb.POST;
+                bool condtionalUpdate = requestMethod == HTTPVerb.PUT || requestMethod == HTTPVerb.DELETE;
```

Comment:
We don't support conditional delete yet

Reply:
Yes. But was accounting the possiblity that we might support it in future. But perhaps not a good idea for now. I will remove it.